sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/ClientRetryPolicy.java

```diff
@@ -121,6 +131,23 @@ public ClientRetryPolicy(GlobalEndpointManager globalEndpointManager,
         return this.throttlingRetry.shouldRetry(e);
     }
 
+    private Mono<ShouldRetryResult> shouldRetryQueryPlanAndAddress() {
```

Comment:
The logic is not query/address dependent, can it be named generic to convey the intent.

Reply:
This is very dependent on query/address retry logic, we can revisit it when we will do other meta data calls. This is good for readability.